aten/src/ATen/native/cuda/Blas.cpp

```diff
@@ -656,11 +656,14 @@ inline void dot_check(const Tensor& self, const Tensor& other) {
 
 } // anonymous namespace
 
-Tensor dot_cuda(const Tensor& self, const Tensor& other) {
+TORCH_IMPL_FUNC(dot_out_cuda)(const Tensor& self, const Tensor& other, const Tensor& result) {
   if (self.is_complex()) {
     if (self.is_conj()) {
       if (other.is_conj()) {
-        return (dot_cuda(self.conj(), other.conj())).conj();
+        auto temp_result = at::empty({}, self.options());
+        at::dot_out(temp_result, self.conj(), other.conj());
+        result.fill_(temp_result.conj());
```

Comment:
Does this change the performance characteristics of this kernel? Could you please provide some profiling results?

Reply:
You can follow the instructions here: https://github.com/pytorch/pytorch/blob/main/CONTRIBUTING.md#cuda-development-tips (I already see below you have done something similar for the CPU path).

Reply:
> If you are working on the CUDA code, 
Except we're not working on CUDA code here and we don't have access to build machines with CUDA HW.

TBH it doesn't matter if we did: if this part of the code is indeed performance critical then you must have a test suite and a workflow that regularly checks for regressions, right? (That's why I was asking which workflow we need to trigger against this PR)
If you don't it's probably easier to do like for other PRs: merge it and revert if it causes issues in the nightlies?